iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/gcp/IamResourcesCheckPart.java

```diff
@@ -0,0 +1,157 @@
+package org.sonar.iac.terraform.checks.gcp;
+
+import org.sonar.iac.common.api.checks.CheckContext;
+import org.sonar.iac.common.api.tree.TextTree;
+import org.sonar.iac.common.checks.PropertyUtils;
+import org.sonar.iac.common.checks.TextUtils;
+import org.sonar.iac.terraform.api.tree.AttributeTree;
+import org.sonar.iac.terraform.api.tree.BlockTree;
+import org.sonar.iac.terraform.checks.AbstractResourceCheck;
+
+
+public class IamResourcesCheckPart extends AbstractResourceCheck {
+
+  private static final String MESSAGE = "Make sure that assigning the %s role is safe here.";
+
+  private static final String[] PRIVILEGED_ROLES = {"ADMIN", "MANAGER", "OWNER", "SUPERUSER"};
+
+  @Override
+  protected void registerResourceChecks() {
+    register(IamResourcesCheckPart::checkIamResources, IAM_RESOURCE_NAMES);
+  }
+
+  private static void checkIamResources(CheckContext ctx, BlockTree resource) {
+    PropertyUtils.get(resource, "role", AttributeTree.class)
+      .filter(attr -> TextUtils.matchesValue(attr.value(), IamResourcesCheckPart::isaPrivilegedRole).isTrue())
+      .ifPresent(attr -> ctx.reportIssue(attr, String.format(MESSAGE, ((TextTree) attr.value()).value())));
+  }
+
+  private static boolean isaPrivilegedRole(String roleName) {
+    // Bear in mind PRIVILEGED_ROLES are all CAPS:
+    roleName = roleName.toUpperCase();
+    for (String privilegedRole : PRIVILEGED_ROLES) {
+      if (roleName.contains(privilegedRole)) {
+        return true;
+      }
+    }
+    return false;
+  }
```

Comment:
Fine for me, but instead of a loop calling 4 times contains and toUpperCase on the string. I would simply create a small predicate based on a pattern and use Pattern.CASE_INSENSITIVE as a flag.